src/App.tsx

```diff
@@ -14,11 +18,92 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  showGoods = () => {
+    const { selectedGoods } = this.state;
+
+    if (selectedGoods.length === 0) {
+      return 'No goods selected';
+    }
+
+    if (selectedGoods.length === 1) {
+      return `${selectedGoods[0]} is selected`;
+    }
+
+    const copyOfGoods = [...selectedGoods];
+    const lastItem = copyOfGoods.pop();
+
+    return `${copyOfGoods.join(', ')} and ${lastItem} are selected`;
+  };
+
+  clear = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  addGood = (goodItem: string) => {
+    this.setState(state => ({
+      selectedGoods: [...state.selectedGoods, goodItem],
+    }));
+  };
+
+  deleteGood = (goodItem: string) => {
+    this.setState(state => ({
+      selectedGoods: state.selectedGoods.filter(item => (
+        item !== goodItem
+      )),
+    }));
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="app">
+        {selectedGoods.length > 0
+          ? (
+            <div className="title">
+              <h1 className="title__name">
+                {this.showGoods()}
+              </h1>
+              <button
+                type="button"
+                className="title__btn"
+                onClick={this.clear}
+              >
+                Clear
+              </button>
+            </div>
+          )
+          : <h1 className="title__name">No goods selected</h1>}
+
+        <ul className="goods-list">
+          {goodsFromServer.map((good) => (
+            <li className="goods-list__item" key={good}>
+              {good}
+              <button
+                className="goods-list__btn"
+                type="button"
+                onClick={() => {
+                  // eslint-disable-next-line @typescript-eslint/no-unused-expressions
```

Comment:
this is not right. Mb move this to other function

Reply:
I'll use if...else instead of ternary operator. Bacause I didn't understand why linter show me this proplem when I used ternary operator, so i desided to use quick fix)